src/server/ua_subscription_alarms_conditions.c

```diff
@@ -1491,6 +1491,7 @@ confirmMethodCallback(UA_Server *server, const UA_NodeId *sessionId,
 
 static UA_StatusCode
 setRefreshMethodEventFields(UA_Server *server, const UA_NodeId *refreshEventNodId) {
+    UA_LOCK_ASSERT(&server->serviceMutex, 1);
```

Comment:
As a rule, if a method is "inside" (with the serviceMutext taken), then it should not call any public methods starting with `UA_Server`.

As a clean solution, split out from `UA_Server_setConditionField` an internal method `setConditionField`.
So `UA_Server_setConditionField` only takes the lock and then calls the internal method.
An you can add the `UA_THREADSAFE` attribute to `UA_Server_setConditionField` in the public API.

With that in place, use `setConditionField` directly from `setRefreshMethodEventFields`.

That same pattern applies everywhere.